toolchain/sem_ir/inst_kind.def

```diff
@@ -11,14 +11,31 @@
 // - CARBON_SEM_IR_INST_KIND(Name)
 //   Invoked for each kind of semantic instruction.
 //
-// The invocation of the above macro will be wrapped in one of these, which by
-// default expand to their argument:
+// The invocation of the above macro will be wrapped in one macro from each of
+// the following sets, which by default expand to their argument:
+//
+// Whether the instruction can define a type:
 // - CARBON_SEM_IR_INST_KIND_TYPE(...)
-//   Invoked for each instruction that might define a type constant.
+//   Invoked for each instruction that is always of type `type`, and might
+//   define a type constant.
+// - CARBON_SEM_IR_INST_KIND_MAYBE_TYPE(...)
```

Comment:
The CONSTANT macros seem to be using a prefix, whereas TYPE is suffixed.

WDYT of MAYBE_TYPE -> TYPE_IN_CONSTANT or TYPE_CONDITIONAL, NOT_TYPE -> TYPE_NEVER (consistent with CONSTANT_NEVER)?

Reply:
Sounds good, I'll send a separate PR.